runtime/interpreter/decode.go

```diff
@@ -294,14 +294,14 @@ func (d StorableDecoder) decodeStorable() (atree.Storable, error) {
 	return storable, nil
 }
 
-func (d StorableDecoder) decodeCharacter(v string) (CharacterValue, error) {
+func (d StorableDecoder) decodeCharacter(memoryGauge common.MemoryGauge, v string) (CharacterValue, error) {
 	if !sema.IsValidCharacter(v) {
 		return "", fmt.Errorf(
 			"invalid character encoding: %s",
 			v,
 		)
 	}
-	return NewCharacterValue(v), nil
+	return NewCharacterValue(memoryGauge, v), nil
```

Comment:
Good point!

Reply:
I like this idea but doing so would require changing `DecodeStorable` in `decode.go` and possibly its callers (they all pass `nil` so far).

An alternative is to change `UseConstantMemory` to be a function acting instead of a method:

```

// UseConstantMemory uses a pre-determined amount of memory
//
func UseConstantMemory(memoryGauge common.MemoryGauge, kind common.MemoryKind) {
	memoryGauge.UseMemory(common.MemoryUsage{
		Kind:   kind,
		Amount: 1,
	})
}
```

That makes sense anyway IMO since every kind of MemoryGauge will handle constants the same way.

If there're no objections, I'll convert `UseConstantMemory` to a function.